src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/AllocationQueryTest.kt

```diff
@@ -0,0 +1,82 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.integration
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserQualification
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserRole
+
+class AllocationQueryTest : IntegrationTestBase() {
+  @Autowired
+  lateinit var realUserRepository: UserRepository
+
+  @Test
+  fun `findQualifiedAssessorWithLeastPendingAllocations query works as described`() {
```

Comment:
I think this test is good, but I'm finding it a bit hard to follow - perhaps with some `given/when/then` type comments to make it clear what is happening in the setup?